compressors/huffman_coder.py

```diff
@@ -0,0 +1,133 @@
+from dataclasses import dataclass
+from typing import Any
+from core.data_stream import BitsDataStream, DataStream
+from core.prob_dist import ProbabilityDist
+from core.data_compressor import DataCompressor
+from core.data_transformer import (
+    BitstringToBitsTransformer,
+    CascadeTransformer,
+    BitsParserTransformer,
+    LookupTableTransformer,
+)
+
+
+@dataclass
+class HuffmanTree:
```

Comment:
should we call this class HuffmanNode, and have another class called HuffmanTree with a single variable called RootNode? That would be semantically more correct, but perhaps adds a bit more complexity

Reply:
Makes sense. Let me try to fix this